cli/src/user.rs

```diff
@@ -340,14 +393,60 @@ impl User {
                     return Err("error".to_string());
                 }
             };
+
+            let processed_message_credential: Credential = processed_message.credential().clone();
+
             match processed_message.into_content() {
                 ProcessedMessageContent::ApplicationMessage(application_message) => {
-                    let application_message =
-                        String::from_utf8(application_message.into_bytes()).unwrap();
+                    let sender_name = match self
+                        .contacts
+                        .get(processed_message_credential.identity())
+                    {
+                        Some(c) => c.username.clone(),
+                        None => {
+                            // Contact list is not updated right now, get the identity from the mls_group member
+                            let mut user_id: String = String::new();
+                            for Member {
+                                index: _,
+                                encryption_key: _,
+                                signature_key,
+                                credential,
+                            } in mls_group.members()
+                            {
+                                if self
+                                    .identity
+                                    .borrow()
+                                    .credential_with_key
+                                    .signature_key
+                                    .as_slice()
+                                    != signature_key.as_slice()
+                                {
+                                    if processed_message_credential.identity()
+                                        == credential.identity()
+                                    {
+                                        log::debug!("update::Processing ApplicationMessage read sender name from credential identity for group {} ", group.group_name);
+                                        user_id = str::from_utf8(credential.identity())
+                                            .unwrap()
+                                            .to_owned()
+                                    }
+                                }
+                            }
+                            user_id
+                        }
+                    };
+                    if sender_name.is_empty() {
+                        panic!("We received a message from an unknown member group !");
+                    }
```

Comment:
If a member chooses an empty string as their identity, then that should be okay, so no need to panic here.

Reply:
done